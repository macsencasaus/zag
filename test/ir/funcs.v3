// RUN: %vector -emit-ir %s

fn main() i32 {
    var x: i32 = 2;
    var y: i32 = 1, z: i32 = 3;
    var aa = 3;
    var bb = 0;
    var zz = 1, yy = 10;
    return 0;
}

fn id(x: i64) i64 {
    return x; 
}

fn three() i64 {
    return 3;
}

fn stack_alignment() i32 {
    var a: i8 = 0;
    var b: i8 = 1;
    var c: i16 = 2;
    var d: i16 = 3;
    var aa: i8 = 0;
    var e: i32 = 4;
    var f: i32 = 5;
    var bb: i8 = 0;
    var g: i64 = 6;
    var h: i64 = 7;
    return 0;
}

fn func_ptrs() i32 {
    var this_fn: fn() i32 = func_ptrs;
    return 0;
}

fn ptrs() i32 {
    var x = 3;
    var p: *i64 = &x;
    var y = *p;
    *p = 2;

    return 0;
}
