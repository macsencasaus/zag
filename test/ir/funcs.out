ir/funcs.v3:3:8: error: Expected token type '(', but got identifier
